File: backend/tomato/src/main/java/com/lelin/tomato/model/Task.java
package com.lelin.tomato.model;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String title;

  private String description;

  @Builder.Default
  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private Priority priority = Priority.MEDIUM;

  @Column(nullable = false)
  private LocalDateTime createdAt;

  // Due time for time bomber
  private LocalDateTime dueTime;

  private LocalDateTime completedAt;

  // Has the task missed its dueTime?
  @Column(nullable = false)
  private boolean expired;

  // Whether time bomber should track this
  @Column(nullable = false)
  private boolean timeBombEnabled;

  @Column(nullable = false)
  private boolean completed;

  @Column(nullable = false)
  private Long userId;

  @Column(nullable = false)
  private int tomatoesEarned = 0;
}

File: mobile/tomato-mobile/src/types/Task.ts
export type Priority = 'LOW' | 'MEDIUM' | 'HIGH';

export interface Task {
  id: number;
  title: string;
  description: string | null;
  priority: Priority;
  dueTime: string | null;
  expired: boolean;
  timeBombEnabled: boolean;
  completed: boolean;
  tomatoesEarned: number;
  createdAt: string;
  completedAt: string | null;
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  priority?: Priority;
  dueTime?: string;
  timeBombEnabled?: boolean;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string | null;
  priority?: Priority;
  dueTime?: string | null;
  timeBombEnabled?: boolean;
  completed?: boolean;
}
